packages/framework/ai-collab/src/test/ai-collab/debugLog.spec.ts

```diff
@@ -0,0 +1,423 @@
+	title: sf.string,
+	tasks: sf.array(
+		sf.object("Task", {
+			title: sf.string,
+			description: sf.string,
+		}),
+	),
+}) {}
+
+const factory = SharedTree.getFactory();
+
+const OPENAI_API_KEY = ""; // DON'T COMMIT THIS
+
+describe.skip("Debug Log Works as expected", () => {
+	let tree: ITree;
+	let view: TreeView<typeof TestAppSchema>;
+	const debugLog: DebugEvent[] = [];
+	let aiCollabResponse: AiCollabSuccessResponse | AiCollabErrorResponse;
+
+	before(async function () {
+		this.timeout(20000);
+		tree = factory.create(
+			new MockFluidDataStoreRuntime({ idCompressor: createIdCompressor() }),
+			"tree",
+		);
+		view = tree.viewWith(new TreeViewConfiguration({ schema: TestAppSchema }));
+		view.initialize({
+			title: "This is a group of tasks",
+			tasks: [
+				{
+					title: "Task 1",
+					description: "This is the first task",
+				},
+				{
+					title: "Task 2",
+					description: "This is the second task",
+				},
+			],
+		});
+
+		aiCollabResponse = await aiCollab({
+			openAI: {
+				client: new OpenAI({
+					apiKey: OPENAI_API_KEY,
+				}),
+				modelName: "gpt-4o",
+			},
+			treeNode: view.root,
+			prompt: {
+				systemRoleContext: "You're a helpful AI assistant",
+				userAsk:
+					"Change the title to 'Hello World', remove the existing tasks and then create two new placeholder tasks",
+			},
+			limiters: {
+				maxModelCalls: 10,
+			},
+			// Notice this is our naive implementation of a debug log handler. We can use this to assert the events are as expected.
+			debugEventLogHandler: (event) => {
+				console.log(`Received event: ${event.eventName}`);
+				if (
+					event.eventName === "APPLIED_EDIT_SUCCESS" ||
+					event.eventName === "APPLIED_EDIT_FAILURE"
+				) {
+					console.log(
+						`${
+							event.eventName === "APPLIED_EDIT_SUCCESS"
+								? "Succesfully applied"
+								: "Failed to appply"
+						} tree edit: ${JSON.stringify(
+							(event as unknown as ApplyEditSuccess).edit,
+							// eslint-disable-next-line unicorn/no-null
+							null,
+							2,
+						)}`,
+					);
+				}
+				debugLog.push(event);
+			},
+			planningStep: true,
+			finalReviewStep: true,
+		});
+
+		// confirming we get a successful response with a filled debug log.
+		assert.strictEqual(aiCollabResponse.status, "success");
+		assert.strictEqual(debugLog.length > 0, true);
+	});
+
+	const assertDebugEventCoreInterfaceIsValid = (event: DebugEvent | undefined): void => {
+		assert(event !== undefined);
+		assert(event.id !== undefined);
+		assert(event.timestamp !== undefined);
+		assert(event.traceId !== undefined);
+	};
+
+	it("Contains the core event loop started and ended events", () => {
+		const expectedTraceId = (debugLog[0] as DebugEvent).traceId; // All debug events from the same execution should have the same trace id
+		assert(expectedTraceId !== undefined);
+
+		const debugEvent1: CoreEventLoopStarted = debugLog[0] as CoreEventLoopStarted;
+
+		assertDebugEventCoreInterfaceIsValid(debugEvent1);
```

Comment:
These calls feel redundant when we have to do `deepStrictEqual` on events anyway.